static/src/js/containers/DatepickerContainer/DatepickerContainer.jsx

```diff
@@ -35,13 +36,35 @@ class DatepickerContainer extends Component {
 
   /**
    * Set view back to the default when a user closes the datepicker
+   * Also set the datetime to be the endOf or startOf day depending on the type
    */
-  onBlur() {
-    const { viewMode } = this.props
+  onBlur(value) {
+    const {
+      format,
+      onSubmit,
+      type,
+      viewMode
+    } = this.props
 
     this.picker.current.setState({
       currentView: viewMode
     })
+
+    let valueToSet = null
+
+    if (typeof value !== 'string' && moment.isMoment(value) && !isCustomTime(value)) {
+      if (type === 'start') {
+        valueToSet = value.startOf('day')
+      } else if (type === 'end') {
+        valueToSet = value.endOf('day')
+      } else {
+        valueToSet = value
+      }
+    } else {
+      valueToSet = moment.utc(value, format, true)
+    }
+
+    onSubmit(valueToSet)
```

Comment:
We could add a `expect(props.onSubmit).toHaveBeenCalledTimes(1)` to the `onBlur` test in enzyme to get some coverage. I understand the `onInput` function is a bit tricker since I don't think we have access to the `event` value in that one I think when we make this RTL that will be easier